x-pack/plugins/alerting/server/lib/is_rule_snoozed.ts

```diff
@@ -6,47 +6,60 @@
  */
 
 import { RRule, ByWeekday, Weekday, rrulestr } from 'rrule';
+import { first } from 'lodash';
 import { SanitizedRule, RuleTypeParams } from '../../common/rule';
 
 type RuleSnoozeProps = Pick<SanitizedRule<RuleTypeParams>, 'muteAll' | 'snoozeSchedule'>;
 
-export function getRuleSnoozeEndTime(rule: RuleSnoozeProps): Date | null {
+export function getActiveSnoozes(
+  rule: RuleSnoozeProps
+): Array<{ snoozeEndTime: Date; id: string }> | null {
   if (rule.snoozeSchedule == null) {
     return null;
   }
 
   const now = Date.now();
-  for (const snooze of rule.snoozeSchedule) {
-    const { duration, rRule } = snooze;
-    const startTimeMS = Date.parse(rRule.dtstart);
-    const initialEndTime = startTimeMS + duration;
-    // If now is during the first occurrence of the snooze
-
-    if (now >= startTimeMS && now < initialEndTime) return new Date(initialEndTime);
-
-    // Check to see if now is during a recurrence of the snooze
-    if (rRule) {
-      try {
-        const rRuleOptions = {
-          ...rRule,
-          dtstart: new Date(rRule.dtstart),
-          until: rRule.until ? new Date(rRule.until) : null,
-          wkst: rRule.wkst ? Weekday.fromStr(rRule.wkst) : null,
-          byweekday: rRule.byweekday ? parseByWeekday(rRule.byweekday) : null,
+  return rule.snoozeSchedule
+    .map((snooze) => {
```

Comment:
I think it makes it marginally more computationally efficient at the expense of readability, because reducing an array into another array is weird. We're unlikely to have a situation where we're dealing with 100,000 entries in this array that it's iterating over so it'd have a negligible effect on speed in the real world.